digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala

```diff
@@ -83,8 +83,17 @@ class ProjectValidator {
     }
   }
 
-  def checkNotReserved(number: Long, part: Long, currentProject: Project): Unit = {
-    val project = projectReservedPartDAO.roadPartReservedByProject(number, part, currentProject.id, withoutProjectId = true)
+  def checkReservedPartInProject(number: Long, part: Long, currentProject: Project, linkStatus: LinkStatus): Unit = {
```

Comment:
Is it possible to pass LinkStatus as optional parameter and use just one checkReservedPartInOtherProject? And combine ProjectValidationExceptions like:
 if (project.isEmpty) { throw new ProjectValidationException(RoadPartNotReservedInProjectMessage.format(number, part, currentProject.name))
} else {
throw new ProjectValidationException(RoadReservedOtherProjectMessage.format(number, part, currentProject.name)) 
}

Reply:
well ive refactored the DAOs fetch methods a bit but the val project are not the opposite in the functions and the if else will not work n this case